backend/btrixcloud/colls.py

```diff
@@ -391,10 +581,228 @@ async def add_successful_crawl_to_collections(self, crawl_id: str, cid: UUID):
             )
             await self.update_crawl_collections(crawl_id)
 
+    async def get_org_public_collections(
+        self,
+        org_slug: str,
+        page_size: int = DEFAULT_PAGE_SIZE,
+        page: int = 1,
+        sort_by: Optional[str] = None,
+        sort_direction: int = 1,
+    ):
+        """List public collections for org"""
+        try:
+            org = await self.orgs.get_org_by_slug(org_slug)
+        # pylint: disable=broad-exception-caught
+        except Exception:
+            # pylint: disable=raise-missing-from
+            raise HTTPException(status_code=404, detail="public_profile_not_found")
+
+        if not org.enablePublicProfile:
+            raise HTTPException(status_code=404, detail="public_profile_not_found")
+
+        collections, _ = await self.list_collections(
+            org,
+            page_size=page_size,
+            page=page,
+            sort_by=sort_by,
+            sort_direction=sort_direction,
+            public_colls_out=True,
+        )
+
+        public_org_details = PublicOrgDetails(
+            name=org.name,
+            description=org.publicDescription or "",
+            url=org.publicUrl or "",
+        )
+
+        return OrgPublicCollections(org=public_org_details, collections=collections)
+
+    async def list_urls_in_collection(
+        self,
+        coll_id: UUID,
+        oid: UUID,
+        url_prefix: Optional[str] = None,
+        page_size: int = DEFAULT_PAGE_SIZE,
+        page: int = 1,
+    ) -> Tuple[List[PageUrlCount], int]:
+        """List all URLs in collection sorted desc by snapshot count"""
```

Comment:
Do we mean list all pages in collection, right?

Reply:
It's actually URLs, not pages. Our "pages" are snapshots (URL + timestamp combo specific to a crawl), so we may have multiple pages/snapshots for the same URL in a collection

Reply:
Hm, maybe we should call these Page URLs instead of just URLs, as that may imply all resources, including non pages?